### PR TITLE
Implement listing AutomaticScenarioAssignments in Director API

### PR DESCRIPTION
**Pull request by pkosiec**
**Description**

Changes proposed in this pull request:

- Implement listing AutomaticScenarioAssignments in Director API

**Related issue(s)**
https://github.com/kyma-incubator/compass/issues/1026

### CODE REVIEW

File: components/director/internal/domain/scenarioassignment/resolver_test.go [hunk 1]
```diff
@@ -264,3 +266,121 @@ func TestResolver_AutomaticScenarioAssignmentForSelector(t *testing.T) {
...
+			},
+			ConverterFn: func() *automock.Converter {
+				conv := &automock.Converter{}
+				return conv
+			},
+			ExpectedResult: nil,
+			ExpectedErr:    testErr,
+		},
+		{
+			Name:            "Returns error when Packages listing failed",
```
Comment (pkosiec) [inline components/director/internal/domain/scenarioassignment/resolver_test.go]:
```suggestion
			Name:            "Returns error when Assignments listing failed",
```

File: components/director/internal/domain/scenarioassignment/service_test.go [hunk 2]
```diff
@@ -128,3 +131,101 @@ func TestService_GetForSelector(t *testing.T) {
...
+			RepositoryFn: func() *automock.Repository {
+				repo := &automock.Repository{}
+				return repo
+			},
+			PageSize:           101,
+			ExpectedResult:     &modelPage,
+			ExpectedErrMessage: "page size must be between 1 and 100",
+		},
+		{
+			Name: "Returns error when Package listing failed",
```
Comment (pkosiec) [inline components/director/internal/domain/scenarioassignment/service_test.go]:
```suggestion
			Name: "Returns error when Assignments listing failed",
```

File: components/director/internal/domain/scenarioassignment/repository.go [hunk 3]
```diff
@@ -84,3 +86,24 @@ func (r *repository) GetForScenarioName(ctx context.Context, tenantID, scenarioN
 
 	return assignmentModel, nil
 }
+
+func (r *repository) List(ctx context.Context, tenantID string, pageSize int, cursor string) (*model.AutomaticScenarioAssignmentPage, error) {
+	var collection EntityCollection
+	page, totalCount, err := r.pageableQuerier.List(ctx, tenantID, pageSize, cursor, "scenario", &collection)
```
Comment (crabtree) [inline components/director/internal/domain/scenarioassignment/repository.go]:
```suggestion
	page, totalCount, err := r.pageableQuerier.List(ctx, tenantID, pageSize, cursor, scenarioColumn, &collection)
```

File: components/director/internal/domain/scenarioassignment/repository_test.go [hunk 4]
```diff
@@ -170,3 +172,82 @@ func TestRepositoryGetForSelector(t *testing.T) {
...
+		assert.Equal(t, scenarioName1, modelAssignment.Data[0].ScenarioName)
+		assert.Equal(t, scenarioName2, modelAssignment.Data[1].ScenarioName)
+		assert.Equal(t, "", modelAssignment.PageInfo.StartCursor)
+		assert.Equal(t, totalCount, modelAssignment.TotalCount)
+		convMock.AssertExpectations(t)
+		sqlMock.AssertExpectations(t)
+	})
+
+	t.Run("DB Error", func(t *testing.T) {
+		// given
```
Comment (crabtree) [inline components/director/internal/domain/scenarioassignment/repository_test.go]: // GIVEN

File: components/director/internal/domain/scenarioassignment/repository_test.go [hunk 5]
```diff
@@ -170,3 +172,82 @@ func TestRepositoryGetForSelector(t *testing.T) {
...
+		repo := scenarioassignment.NewRepository(nil)
+		sqlxDB, sqlMock := testdb.MockDatabase(t)
+		testError := errors.New("test error")
+
+		sqlMock.ExpectQuery(selectQuery).
+			WithArgs(tenantID).
+			WillReturnError(testError)
+		ctx := persistence.SaveToContext(context.TODO(), sqlxDB)
+
+		// when
```
Comment (crabtree) [inline components/director/internal/domain/scenarioassignment/repository_test.go]:
```suggestion
		// WHEN
```

File: components/director/internal/domain/scenarioassignment/repository_test.go [hunk 6]
```diff
@@ -170,3 +172,82 @@ func TestRepositoryGetForSelector(t *testing.T) {
...
+		modelAssignment, err := repo.List(ctx, tenantID, inputPageSize, inputCursor)
+
+		// then
```
Comment (crabtree) [inline components/director/internal/domain/scenarioassignment/repository_test.go]:
```suggestion
		// THEN
```

File: components/director/internal/domain/scenarioassignment/resolver.go [hunk 7]
```diff
@@ -144,3 +133,43 @@ func (r *Resolver) AutomaticScenarioAssignmentForSelector(ctx context.Context, i
...
+	if err != nil {
+		return nil, err
+	}
+	defer r.transact.RollbackUnlessCommited(tx)
+
+	ctx = persistence.SaveToContext(ctx, tx)
+
+	page, err := r.svc.List(ctx, *first, cursor)
+	if err != nil {
+		return nil, err
```
Comment (crabtree) [inline components/director/internal/domain/scenarioassignment/resolver.go]: Please wrap the errors, to be consistent with other implementations

File: components/director/internal/domain/scenarioassignment/repository_test.go [hunk 8]
```diff
@@ -162,15 +164,94 @@ func TestRepositoryGetForSelector(t *testing.T) {
...
+			WillReturnRows(rows)
+
+		sqlMock.ExpectQuery(countQuery).
+			WithArgs(tenantID).
+			WillReturnRows(testdb.RowCount(2))
+
+		ctx := persistence.SaveToContext(context.TODO(), sqlxDB)
+		convMock := &automock.EntityConverter{}
+		convMock.On("FromEntity", ent1).Return(mod1, nil)
+		convMock.On("FromEntity", ent2).Return(mod2, nil)
```
Comment (dbadura) [inline components/director/internal/domain/scenarioassignment/repository_test.go]: Why do you return (mode, *nil*)? 
`return(model)` is sufficient

File: components/director/internal/domain/scenarioassignment/service.go [hunk 9]
```diff
@@ -65,3 +66,16 @@ func (s *service) GetForScenarioName(ctx context.Context, scenarioName string) (
...
 	return sa, nil
 }
+
+func (s *service) List(ctx context.Context, pageSize int, cursor string) (*model.AutomaticScenarioAssignmentPage, error) {
+	tnt, err := tenant.LoadFromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	if pageSize < 1 || pageSize > 100 {
```
[the remark this answers is not in the document]
Comment (pkosiec) [reply components/director/internal/domain/scenarioassignment/service.go]: Personally I like variable on the left side of the statement 🙂

**Pull request merged**